BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m

```diff
@@ -0,0 +1,104 @@
+//  BoxContentSDK
+//
+
+#import "BOXRequest_Private.h"
+#import "BOXUserAvatarRequest.h"
+
+#import "BOXAPIDataOperation.h"
+
+@interface BOXUserAvatarRequest ()
+
+@property (nonatomic, readonly, strong) NSOutputStream *outputStream;
+
+@end
+
+@implementation BOXUserAvatarRequest
+
+- (instancetype)initWithUserID:(NSString *)userID
+{
+    if (self = [super init]) {
+        _userID = userID;
+        _outputStream = [[NSOutputStream alloc] initToMemory];
+
+        _avatarType = BOXAvatarTypeUnspecified;
+    }
+
+    return self;
+}
+
+- (BOXAPIOperation *)createOperation
+{
+    NSURL *URL = [self URLWithResource:BOXAPIResourceUsers
+                                    ID:self.userID
+                           subresource:BOXAPISubresourceAvatar
+                                 subID:nil];
+
+    NSMutableDictionary *queryParameters = [NSMutableDictionary dictionary];
+
+    NSString *avatarTypeString = nil;
+    switch (self.avatarType) {
+        case BOXAvatarTypeSmall:
+            avatarTypeString = @"small";
+            break;
+        case BOXAvatarTypeLarge:
+            avatarTypeString = @"large";
+            break;
+        case BOXAvatarTypeProfile:
+            avatarTypeString = @"profile";
+            break;
+        case BOXAvatarTypeUnspecified:
+        default:
+            break;
+    }
+    if ([avatarTypeString length] > 0) {
+        queryParameters[BOXAPIParameterKeyAvatarType] = avatarTypeString;
+    }
+
+    BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:URL
+                                                         HTTPMethod:BOXAPIHTTPMethodGET
+                                              queryStringParameters:queryParameters
+                                                     bodyDictionary:nil
+                                                       successBlock:nil
+                                                       failureBlock:nil];
+    dataOperation.modelID = self.userID;
+    dataOperation.outputStream = self.outputStream;
+    dataOperation.isSmallDownloadOperation = YES;
+
+    return dataOperation;
+}
+
+- (void)performRequestWithProgress:(BOXProgressBlock)progressBlock completion:(BOXImageBlock)completionBlock
+{
+    if (completionBlock) {
+        BOOL isMainThread = [NSThread isMainThread];
+        BOXAPIDataOperation *dataOperation = (BOXAPIDataOperation *)self.operation;
+
+        if (progressBlock) {
+            dataOperation.progressBlock = ^(long long expectedTotalBytes, unsigned long long bytesReceived) {
+                [BOXDispatchHelper callCompletionBlock:^{
+                    progressBlock(bytesReceived, expectedTotalBytes);
+                } onMainThread:isMainThread];
+            };
+        }
+
+        NSOutputStream *outputStream = self.outputStream;
+        dataOperation.successBlock = ^(NSString *modelID, long long expectedTotalBytes) {
+            NSData *data = [outputStream propertyForKey:NSStreamDataWrittenToMemoryStreamKey];
```

Comment:
Possibly. The information about 'large' and 'profile' types is not very clear. However this is highly unlikely right now as the Web UI prevents you from uploading a profile image that is greater than 1MB. It is safe for now to assume these are small enough (treating them the same as file thumbnails).